test/test_block_store_service.py

```diff
@@ -116,7 +116,7 @@ def test_volume_properties(self):
                 self.assertEqual(test_vol.attachments.instance_id,
                                  test_instance.id)
                 if (self.provider.PROVIDER_ID != 'azure' and
-                    self.provider.PROVIDER_ID != 'gce'):
+                        self.provider.PROVIDER_ID != 'gce'):
```

Comment:
You can programmatically call vol.attachments.device; however, attached devices on GCE are mounted like /dev/disk/by-id/google-*. It will not be /dev/sda2, as the test expects.

Reply:
GCE doesn't allow you to specify an explicit path, though you can specify a device name.
When you specify the name, then you can assume an explicit path `/dev/disk/by-id-google-<name>` is created. Underneath this is a link to `/dev/sda<char>` where the characters start with `a` for the first disk (the boot disk) and `b`, `c`, etc. for subsequent disks.
See https://medium.com/@DazWilkin/compute-engine-identifying-your-devices-aeae6c01a4d7.

However - Is the ability to specify a specific device path important for the volume API?
If so, then I think it would be preferred for the create() API to accept a device name and then have an implementation-specific device path returned.
That said - I'm not familiar with the Azure API and how it would fit here.

Reply:
@chiniforooshan @mbookman That makes sense. It looks like we should change the vol.attach() interface to not accept the device id as a parameter. Is there a suitable way we could generalize this test but still verify that attachments.device remains a reasonable value, and not null/empty or invalid. I suppose we could just have a regex testing for a slash followed by a slash?

Reply:
That's true, the ideal test would have been an ssh based test - but it was never done because it seemed like a lot of work for relatively little gain. I think your suggestion makes sense in the meantime, thanks.